ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -114,18 +114,8 @@ private String generateConnectionId(SocketChannel channel) {
     int localPort = socket.getLocalPort();
     String remoteHost = socket.getInetAddress().getHostAddress();
     int remotePort = socket.getPort();
-    long connectionIdSuffix = IdGenerator.getAndIncrement();
```

Comment:
Isn't `StringBuilder` more efficient than concatenating strings in Java?  Was this just a code readability vs. performance efficiency tradeoff?

Reply:
That is true that `StringBuilder` is more efficient than multiple string concatenations. However, for simple usages of string `+` (single line, no loops, no method call barriers), java will automatically convert this into a `StringBuilder` usage. Intellij suggests replacing such usages of `StringBuilder` with `+`. Inspecting the bytecode for this line,  `StringBuilder` is in fact used:
```
    LINENUMBER 118 L6
    NEW java/lang/StringBuilder
    DUP
    INVOKESPECIAL java/lang/StringBuilder.<init> ()V
    ALOAD 3
    INVOKEVIRTUAL java/lang/StringBuilder.append (Ljava/lang/String;)Ljava/lang/StringBuilder;
    LDC ":"
    INVOKEVIRTUAL java/lang/StringBuilder.append (Ljava/lang/String;)Ljava/lang/StringBuilder;
    ILOAD 4
    INVOKEVIRTUAL java/lang/StringBuilder.append (I)Ljava/lang/StringBuilder;
    LDC "-"
    INVOKEVIRTUAL java/lang/StringBuilder.append (Ljava/lang/String;)Ljava/lang/StringBuilder;
    ALOAD 5
    INVOKEVIRTUAL java/lang/StringBuilder.append (Ljava/lang/String;)Ljava/lang/StringBuilder;
    LDC ":"
    INVOKEVIRTUAL java/lang/StringBuilder.append (Ljava/lang/String;)Ljava/lang/StringBuilder;
    ILOAD 6
    INVOKEVIRTUAL java/lang/StringBuilder.append (I)Ljava/lang/StringBuilder;
    LDC "_"
    INVOKEVIRTUAL java/lang/StringBuilder.append (Ljava/lang/String;)Ljava/lang/StringBuilder;
    LLOAD 7
    INVOKEVIRTUAL java/lang/StringBuilder.append (J)Ljava/lang/StringBuilder;
    INVOKEVIRTUAL java/lang/StringBuilder.toString ()Ljava/lang/String;
    ARETURN
```

Reply:
oh wow cool, didn't know that